internal/controller/server_controller.go

```diff
@@ -285,14 +285,32 @@ func (r *ServerReconciler) handleDiscoveryState(ctx context.Context, log logr.Lo
 	}
 	server.Status.Storages = nil
 	for _, storage := range storages {
-		server.Status.Storages = append(server.Status.Storages, metalv1alpha1.Storage{
-			Name:     storage.Name,
-			Model:    storage.Model,
-			Capacity: resource.NewQuantity(storage.SizeBytes, resource.BinarySI),
-			Type:     string(storage.Type),
-			Vendor:   storage.Vendor,
-			State:    metalv1alpha1.StorageState(storage.State),
-		})
+		a1Storage := metalv1alpha1.Storage{
+			Name:  storage.Name,
+			State: metalv1alpha1.StorageState(storage.State),
+		}
+		for _, drive := range storage.Drives {
+			a1Storage.Drives = append(a1Storage.Drives, metalv1alpha1.StorageDrive{
+				Name:       drive.Name,
+				Model:      drive.Model,
+				Vendor:     drive.Vendor,
+				Capacity:   resource.NewQuantity(drive.SizeBytes, resource.BinarySI),
+				Type:       string(drive.Type),
+				State:      metalv1alpha1.StorageState(drive.State),
+				Rotational: drive.Rotational,
+			})
+		}
+		a1Storage.Volumes = make([]metalv1alpha1.StorageVolume, 0, len(storage.Volumes))
+		for _, volume := range storage.Volumes {
+			a1Storage.Volumes = append(a1Storage.Volumes, metalv1alpha1.StorageVolume{
+				Name:        volume.Name,
+				Capacity:    resource.NewQuantity(volume.SizeBytes, resource.BinarySI),
+				State:       metalv1alpha1.StorageState(volume.State),
+				RAIDType:    string(volume.RAIDType),
+				VolumeUsage: volume.VolumeUsage,
+			})
+		}
+		server.Status.Storages = append(server.Status.Storages, a1Storage)
```

Comment:
Won't this slice grow over time e.g. in a second reconciliation loop?

Reply:
line 286 should reset it